datafusion/optimizer/src/reduce_outer_join.rs

```diff
@@ -438,13 +449,13 @@ mod tests {
                 None,
             )?
             .filter(binary_expr(
-                col("t1.b").gt(lit(10u32)),
+                cast(col("t1.b"), DataType::Int64).gt(lit(10u32)),
```

Comment:
I recommend adding a new test explicitly for the cast case (or at least renaming the test) rather than modifying the existing one so it is clear that the case is covered

Reply:
I'll add some new test cases laster. Thanks!